From b1cd2abf62c7871336b83460abaadfd1b5621fac Mon Sep 17 00:00:00 2001
From: Bogdan Purcareata <bogdan.purcareata@gmail.com>
Date: Mon, 13 Feb 2017 18:40:41 +0200
Subject: [PATCH 084/107] staging: fsl-mc: Add missing header
Git-commit: 07e9ef146071adf316312b3dd4abfe41c9a45a7d
Patch-mainline: v4.11-rc1
References: bsc#1041022

Compiling the fsl-mc bus driver will yield a couple of static analysis
Errors: 
Warning: symbol 'fsl_mc_msi_domain_alloc_irqs' was not declared
Warning: symbol 'fsl_mc_msi_domain_free_irqs' was not declared.
Warning: symbol 'its_fsl_mc_msi_init' was not declared.
Warning: symbol 'its_fsl_mc_msi_cleanup' was not declared.

Since these are properly declared, but the header is not included, add
it in the source files. This way the symbol is properly exported.

Signed-off-by: Bogdan Purcareata <bogdan.purcareata@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Alexander Graf <agraf@suse.de>

---
 drivers/staging/fsl-mc/bus/fsl-mc-msi.c                | 1 +
 drivers/staging/fsl-mc/bus/irq-gic-v3-its-fsl-mc-msi.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/staging/fsl-mc/bus/fsl-mc-msi.c b/drivers/staging/fsl-mc/bus/fsl-mc-msi.c
index 7875f56..5357a39 100644
--- a/drivers/staging/fsl-mc/bus/fsl-mc-msi.c
+++ b/drivers/staging/fsl-mc/bus/fsl-mc-msi.c
@@ -17,6 +17,7 @@
 #include <linux/irqdomain.h>
 #include <linux/msi.h>
 #include "../include/mc-bus.h"
+#include "fsl-mc-private.h"
 
 /*
  * Generate a unique ID identifying the interrupt (only used within the MSI
diff --git a/drivers/staging/fsl-mc/bus/irq-gic-v3-its-fsl-mc-msi.c b/drivers/staging/fsl-mc/bus/irq-gic-v3-its-fsl-mc-msi.c
index 0e2c1b5..87e4471 100644
--- a/drivers/staging/fsl-mc/bus/irq-gic-v3-its-fsl-mc-msi.c
+++ b/drivers/staging/fsl-mc/bus/irq-gic-v3-its-fsl-mc-msi.c
@@ -17,6 +17,7 @@
 #include <linux/of.h>
 #include <linux/of_irq.h>
 #include "../include/mc-bus.h"
+#include "fsl-mc-private.h"
 
 static struct irq_chip its_msi_irq_chip = {
 	.name = "ITS-fMSI",
-- 
1.8.5.6

